### PR TITLE
[Merged by Bors] - refactor(Algebra/Polynomial/Lifts): Golf some proofs

### PR DESCRIPTION
**Pull request by tb65536**
This PR golfs some proofs in `Algebra/Polynomial/Lifts.lean`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Polynomial/Lifts.lean [hunk 1]
```diff
@@ -188,24 +151,14 @@ section Monic
...
-  have p_neq_0 : p ≠ 0 := by intro hp; apply h0; rw [hp]; simp only [natDegree_zero, erase_zero]
-  have hdeg : q.degree < ((X : R[X]) ^ p.natDegree).degree := by
-    rw [@degree_X_pow R, hq.2, ← degree_eq_natDegree p_neq_0]
-    exact degree_erase_lt p_neq_0
-  refine ⟨q + X ^ p.natDegree, ?_, ?_, (monic_X_pow _).add_of_right hdeg⟩
-  · rw [Polynomial.map_add, hq.1, Polynomial.map_pow, map_X, H]
-  · rw [degree_add_eq_right_of_degree_lt hdeg, degree_X_pow, degree_eq_natDegree hp.ne_zero]
+  rw [lifts_iff_coeff_lifts] at hlifts
+  let g : ℕ → R := fun k ↦ (hlifts k).choose
+  have hg : ∀ k, f (g k) = p.coeff k := fun k ↦ (hlifts k).choose_spec
```
Comment (MichaelStollBayreuth) [inline Mathlib/Algebra/Polynomial/Lifts.lean:156]:
```suggestion
  have hg k : f (g k) = p.coeff k := (hlifts k).choose_spec
```

File: Mathlib/Algebra/Polynomial/Lifts.lean [hunk 2]
```diff
@@ -188,24 +151,14 @@ section Monic
...
+  let q : R[X] := X ^ p.natDegree + ∑ k ∈ Finset.range p.natDegree, C (g k) * X ^ k
+  have hq : map f q = p := by simp_rw [q, Polynomial.map_add, Polynomial.map_sum,
+    Polynomial.map_mul, Polynomial.map_pow, map_X, map_C, hg, ← hp.as_sum]
```
Comment (MichaelStollBayreuth) [inline Mathlib/Algebra/Polynomial/Lifts.lean:159]:
```suggestion
  have hq : map f q = p := by
    simp_rw [q, Polynomial.map_add, Polynomial.map_sum, Polynomial.map_mul, Polynomial.map_pow,
      map_X, map_C, hg, ← hp.as_sum]
```
I think this is easier to parse.

**Review by MichaelStollBayreuth — APPROVED** [review]
LGTM!

**Comment by MichaelStollBayreuth** [comment]
maintainer delegate

**Comment by riccardobrasca** [comment]
There are two comments, but LGTM, thanks!

bors d+

**Pull request closed without merging**